src/dlite-type.c

```diff
@@ -1128,21 +1128,21 @@ int dlite_type_scan(const char *src, int len, void *p, DLiteType dtype,
       jsmn_init(&parser);
       r = jsmn_parse(&parser, src, len, tokens, MAX_PROPERTY_TOKENS);
       if (r == JSMN_ERROR_NOMEM)
-        return err(-1, "too many dimensions.  Increase MAX_PROPERTY_TOKENS "
+        return err(dliteIndexError, "too many dimensions.  Increase MAX_PROPERTY_TOKENS "
                    "in dlite-type.c and recompile.");
       else if (r < 0)
-        return err(-1, "cannot parse property: %s: '%s'",
+        return err(dliteTypeError, "cannot parse property: %s: '%s'",
```

Comment:
Wouldn't a dedicated `dliteParseError` be more informative?

Reply:
added